tests/cursor.py

```diff
@@ -164,3 +196,111 @@ def test_uncallable_representation():
 def test_requests_volume_on_iteration(cursor):
     _ = [r for r in cursor]
     assert cursor.client.get.call_count == 1
+
+
+def test_skip_none(mocker):
+    cursor = mock_ordered_cursor(mocker, start=0, count=COLLECTION_SIZE)
+    results = [r for r in cursor]
+    assert len(results) == COLLECTION_SIZE
+    assert results[0]['id'] == 0
+    assert results[-1]['id'] == COLLECTION_SIZE - 1
+    assert cursor.get_page.call_count == 4
+
+
+def test_skip_ten(mocker):
+    skip = 10
+    cursor = mock_ordered_cursor(mocker, start=skip, count=COLLECTION_SIZE)
+    cursor.skip(skip)
```

Comment:
I'm not sure to understand why you `mock_ordered_cursor(..., start=skip, ...)` and then `cursor.skip(skip)`, can you explain?

Reply:
The goal of this test and the ones following is to validate that the cursor is iterating over pages properly taking `skip` and `limit` parmeters into account. These parameters are setting up what is going to be asked of the API, not doing some post-request manipulation of the results.
The `start`, `count` or `factor` arguments given to `mock_ordered_cursor` are used to simulate the expected response from the API. Therefore, we need to align what we want (ie. setting up the cursor) with what is expected to be returned by the `client.get` method that we are patching, feeding `gen_collection` results to it.

I think the tests are ok as is (used them to validate the algo changes) though I admit that they could be a bit better.
One potentially better approach would be to actually monkey patch `request.get` and similar to change the raw results based on the query params of the url requested. I felt this was a bit overkill for what I wanted to achieve here.